src/main/java/controllers/ApplicationController.java

```diff
@@ -49,4 +49,14 @@ public Result sonar(Context context, AttackGameAction g) {
             return Results.badRequest();
         }
     }
+
+    public Result move(Context context, MoveGameAction g) {
+        Game game = g.getGame();
+        boolean result = game.move(g.getDirection());
+        if (result) {
+            return Results.json().render(game);
+        } else {
+            return Results.badRequest();
+        }
```

Comment:
Good job encapsulating the getDirection() within a MoveAction class. This follows object oriented design.